src/de/espend/idea/php/toolbox/type/utils/PhpTypeProviderUtil.java

```diff
@@ -16,46 +17,66 @@
+
+    @Nullable
+    public static String getReferenceSignature(FunctionReference functionReference, char trimKey, Set<Integer> parameterIndexes) {
 
         String refSignature = functionReference.getSignature();
         if(StringUtil.isEmpty(refSignature)) {
             return null;
         }
 
         PsiElement[] parameters = functionReference.getParameters();
-        if (parameters.length != equalParameterCount) {
-            return null;
-        }
 
-        PsiElement parameter = parameters[0];
+        HashMap<Integer, String> parameterSignatures = new HashMap<>();
 
-        // we already have a string value
-        if ((parameter instanceof StringLiteralExpression)) {
-            String param = ((StringLiteralExpression)parameter).getContents();
-            if (StringUtil.isNotEmpty(param)) {
-                return refSignature + trimKey + param;
+        for (int parameterIndex : parameterIndexes) {
+            if (parameterIndex + 1 > parameters.length) {
+                continue;
             }
 
-            return null;
-        }
+            PsiElement parameter = parameters[parameterIndex];
 
-        // whitelist here; we can also provide some more but think of performance
-        // Service::NAME, $this->name and Entity::CLASS;
-        if (parameter instanceof PhpReference && (parameter instanceof ClassConstantReference || parameter instanceof FieldReference)) {
-            String signature = ((PhpReference) parameter).getSignature();
-            if (StringUtil.isNotEmpty(signature)) {
-                return refSignature + trimKey + signature;
+            // we already have a string value
+            if ((parameter instanceof StringLiteralExpression)) {
+                String param = ((StringLiteralExpression)parameter).getContents();
+                if (StringUtil.isEmpty(param)) {
+                    continue;
+                }
+
+                parameterSignatures.put(parameterIndex, param);
+            }
+
+            // whitelist here; we can also provide some more but think of performance
+            // Service::NAME, $this->name and Entity::CLASS;
+            if (parameter instanceof PhpReference && (parameter instanceof ClassConstantReference || parameter instanceof FieldReference)) {
+                String signature = ((PhpReference) parameter).getSignature();
+                if (StringUtil.isEmpty(signature)) {
+                    continue;
+                }
+
+                parameterSignatures.put(parameterIndex, signature);
             }
+        }
 
+        if (parameterSignatures.isEmpty()) {
             return null;
         }
 
-        return null;
+        String parametersSignature = parameterSignatures
+                .entrySet()
+                .stream()
+                .map(entry -> entry.getKey() + entry.getValue())
+                .collect(Collectors.joining(String.valueOf(trimKey)));
+
+        return refSignature + trimKey + trimKey + parametersSignature;
```

Comment:
So it can be split on in [`getBySignature`](https://github.com/Haehnchen/idea-php-toolbox/pull/67/files#diff-670997170f32528da4b4e8563d736a52R89). I know this is weird. I'm joining multiple parameter pairs with the trim key as well, so I couldn't split on that one character alone. 

This `doThing('Foo', 'Bar')` looks like this `{refSignature} ƕƕ 0Foo ƕ 1Bar` (added spaces for clarity).

As I said in the PR description, I'm happy to do it another way. I'm just not sure which characters would be allowed that wouldn't mess up other things.